packages/router/src/router.ts

```diff
@@ -472,8 +472,12 @@ export class Router {
                        };
                      }),
                      switchMap(t => {
+                       const isStateChanged = this.lastSuccessfulNavigation ?
+                           !deepEqual(t.extras.state, this.lastSuccessfulNavigation.extras.state) :
+                           false;
```

Comment:
```suggestion
                       const isStateChanged = 
                           !deepEqual(t?.extras?.state, this.lastSuccessfulNavigation?.extras?.state);
```
If there is no `lasSuccessfulNavigation`, then the state is changed. `this.navigated` will also be `false` so there should really be no difference.

`extras` might actually be `undefined` even though the type doesn't say so. The issue comes from these two places in the code:
https://github.com/angular/angular/blob/cc49a91de7e228c6bdff8ca58936f05abe0ceb9f/packages/router/src/router.ts#L756
https://github.com/angular/angular/blob/cc49a91de7e228c6bdff8ca58936f05abe0ceb9f/packages/router/src/router.ts#L506-L509